### PR TITLE
Propim 480/breadcrumb

### PR DESCRIPTION
**Pull request by Tuxosaurus**
- New Breadcrumb component: a list of links with its spacing and a divider
- Update to HeroSection: add "noMarginBottom" to accomodate the fact that it can be followed directly by the breadcrumb.
_Note: I keep the margin on small devices to make sure there's space between the main action button and the breadcrumb links._

Large device (desktop)
Screen Shot 2020-09-21 at 16 30 28

Small device (mobile)
Screen Shot 2020-09-21 at 16 30 57

Tested locally on Firefox, within ui-library and in the main project using a canary release

### CODE REVIEW

File: src/breadcrumb/Breadcrumb.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export const Breadcrumb = ({ crumbs }: BreadcrumbProps) => (
+  <Fragment>
+    <StyledBreadcrumb itemScope itemType="https://schema.org/BreadcrumbList">
+      {crumbs.map((crumb, index) => {
+        const position = Number(index + 1)
+
+        return (
+          <li
+            key={`crumb-${crumb.name}-${position}`}
```
Comment (wakooka) [inline src/breadcrumb/Breadcrumb.tsx:25]: Why do you add the `index`, isn't the name unique?

File: src/breadcrumb/Breadcrumb.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                <span itemProp="name">{crumb.name}</span>
+              </a>
+
+              {position < crumbs.length && (
+                <span role="separator" className="breadcrumb-separator">
+                  {' > '}
+                </span>
+              )}
+            </TextCaption>
+            <meta itemProp="position" content={`${position}`} />
```
Comment (wakooka) [inline src/breadcrumb/Breadcrumb.tsx:48]:
```suggestion
            <meta itemProp="position" content={position} />

**Review by wakooka — COMMENTED** [review]
If possible I would remove `noMarginBottom` so that we don't accomodate `HeroSection`. If it's too low I would argue that these components need to be in the design system, while they're not we shouldn't create more & more use cases.

**Comment by Tuxosaurus** [comment]
> If possible I would remove `noMarginBottom` so that we don't accomodate `HeroSection`. If it's too low I would argue that these components need to be in the design system, while they're not we shouldn't create more & more use cases.

File: src/breadcrumb/Breadcrumb.tsx (hunk 2 above)
Comment (Tuxosaurus) [reply src/breadcrumb/Breadcrumb.tsx:48]: `content` is required as a string, not a number. I'm already using `position` as a number for this comparison `position < crumbs.length`

File: src/breadcrumb/Breadcrumb.tsx (hunk 1 above)
Comment (Tuxosaurus) [reply src/breadcrumb/Breadcrumb.tsx:25]: `name` isn't enforced as unique, it sounds stupid that 2 links in breadcrumb would be called the same but I prefer to use the `position` for the uniqueness. And I prefer to keep `name` just to make it more readable.

**Comment by Tuxosaurus** [comment]
> If possible I would remove `noMarginBottom` so that we don't accomodate `HeroSection`. If it's too low I would argue that these components need to be in the design system, while they're not we shouldn't create more & more use cases.

Alright I removed it.

File: src/breadcrumb/Breadcrumb.tsx (hunk 2 above)
Comment (wakooka) [reply src/breadcrumb/Breadcrumb.tsx:48]: Oh yes I missed that, then it's a super nit: `content={String(position)} />`

**Review by wakooka — APPROVED** [review]

File: src/breadcrumb/Breadcrumb.tsx (hunk 2 above)
Comment (Tuxosaurus) [reply src/breadcrumb/Breadcrumb.tsx]: Done

**Pull request merged**